### DISCUSSION TITLE
Public folder name clash

### DISCUSSION THREAD

**Discussion by dmudro** [vercel/next.js#15055] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

I want to be able to use [GitLab pages](https://docs.gitlab.com/ee/user/project/pages/) in my Next.js project. GitLab looks the `public` folder specifically:

_GitLab always deploys your website from a very specific folder called public in your repository. When you create a new project in GitLab, a repository becomes available automatically._

The desired content of my desired GL pages is _different_ from my app's public folder so these two folders clash as they both must be named the same.

## Describe the solution you'd like

I wonder if the `public` folder name housing Next's app static assets could be configurable? Similar to the optional `src` directory.

## Describe alternatives you've considered

- put contents of my non Next.js `public` folder into another repository
- wrap Next.js app inside a subfolder of my repo, i. e.:
`root/next-app` with all application code and package.json
`root/public` for non-Next.js public folder

None of these are exactly elegant solutions.

## Additional context

There could be an argument for solving this on GitLab's side, however, I do think Next.js would benefit from configurable name for `public` as that keyword is too generic and perhaps reserved for other tools too.

**Comment by timneutkens** [comment]
Pretty much all frameworks (including ones written in PHP) use `public` in this way so it sounds more like a gitlab bug

**Reply by dmudro** [reply]
Exactly. Many frameworks, tools etc use `public` which why it might be a good idea to make it configurable.

Don't think it's a GitLab _bug_, it's just how they work with hardcoded `public` - the same way Next.js does. 

On a side note, I do find GitHub's pages solution more elegant.
